content/en/blog/improving-observability-opentelemetry-datadog-fission/index.md

```diff
@@ -0,0 +1,288 @@
++++
+title = "Improving Observability with OpenTelemetry & Datadog in Fission"
+date = "2022-01-04T11:30:34+05:30"
+author = "Atulpriya Sharma"
+description = "Guide on how you can improve observability with OpenTelemetry & Datadog in Fission."
+categories = ["Tutorial","Application","Python","Observability"]
+type = "blog"
++++
+
+{{< imgproc opentelemetry-datadog-fission-featured.png Fit "1000X1000" >}}
+Integrating OpenTelemetry & Datadog with Fission functions
+{{< /imgproc >}}
+
+Software development isn’t just about writing a piece of code and running it.
+There are a lot of other processes around it that ensure that your code is going to work as expected in the real world.
+Amongst all such practices, **Observability** is going to be our point of discussion today.
+
+Observability is all about gaining greater control and visibility over your application.
+It actually tells you what is really happening in your application.
+With most of the applications today adopting a microservices-oriented architecture, the applications have become complex with a lot of moving parts.
+And hence observability is a key to build a robust and reliable system.
+**Logs**, **Metrics** and **Trace** are the **3 pillars of observability**. While logs and metrics are very helpful, trace allows you to traverse the entire journey of a request.
+
+There are a handful of applications out there that help you with observability.
+However, one of the major problems with them is that they aren’t vendor agnostic.
+It means that if you want to change your observability tool, you will have to rewrite the code as well.
+That's where **OpenTelemetry** comes into the picture.
+
+## OpenTelemetry & Datadog
+
+OpenTelemetry gives us vendor-neutral APIs that helps us collect and export our traces to any 3rd party application.
+That means that OpenTelemetry will help you collect the traces and send it to any application that you want to by just changing the configuration.
+You can export your traces to applications like Prometheus, Jaeger, Datadog or anything else that you use.
```

Comment:
Lets not mention Prometheus. It is used for metrics not traces.
We can mention Jaeger, Datadog, Signoz, Uptrace etc.

Reply:
Done